e2e/test/DeviceTokenRefreshE2ETests.cs

```diff
@@ -169,16 +172,23 @@ public TestTokenRefresher(string deviceId, string key) : base(deviceId)
                 string deviceId, 
                 string key, 
                 int suggestedTimeToLive, 
-                int timeBufferPercentage) 
+                int timeBufferPercentage,
+                Client.TransportType transport) 
                 : base(deviceId, suggestedTimeToLive, timeBufferPercentage)
             {
                 _key = key;
+                _transport = transport;
             }
 
             protected override Task<string> SafeCreateNewToken(string iotHub, int suggestedTimeToLive)
             {
                 Console.WriteLine($"[{DateTime.UtcNow}] Refresher: Creating new token {_callCount}");
 
+                if (_transport == Client.TransportType.Mqtt)
+                {
+                    suggestedTimeToLive = -4 * 60 - 59; // server side time allowance.
```

Comment:
Can you describe these magic numbers for future reference?

Reply:
I've added a comment on what the 4:59 means. Do you mean we should document how the server applies the 10 minute time allowance in two separate chunks?